### PR TITLE
Fix Mix Precision settings for FSDP Plugins

### LINKED ISSUE #17506: Inconsistent settings for FSDP Precision
### Bug description

The FSDPPrecision class controls the dtype of the model parameters, autocasting and grad scaler.
It has two possible inputs: precision="16-mixed" or "bf16-mixed":
https://github.com/Lightning-AI/lightning/blob/6464650a3c91a75d4a7c4009d08a68a84037a9a1/src/lightning/fabric/plugins/precision/fsdp.py#L32-L34

In all other precision plugins, "mixed" refers to mixed precision training with model weights in float32 and autocasting inputs and operations to lower precision. However, the FSDPPrecision plugin sets the dtype of the model parameters to float16/bfloat16 regardless. This means we are actually running "16-true". 

https://github.com/Lightning-AI/lightning/blob/6464650a3c91a75d4a7c4009d08a68a84037a9a1/src/lightning/fabric/plugins/precision/fsdp.py#L50-L60

Proposal:

For the mixed precision settings, set `param_type=torch.float32`. For the current case, introduce the precision settings "16-true" and "bf16-true". 

### What version are you seeing the problem on?

v1_9

### How to reproduce the bug

_No response_

### Error messages and logs

```
# Error messages and logs here please
```

### Environment

<details>
  <summary>Current environment</summary>

```
#- Lightning Component (e.g. Trainer, LightningModule, LightningApp, LightningWork, LightningFlow):
#- PyTorch Lightning Version (e.g., 1.5.0):
#- Lightning App Version (e.g., 0.5.2):
#- PyTorch Version (e.g., 2.0):
#- Python version (e.g., 3.9):
#- OS (e.g., Linux):
#- CUDA/cuDNN version:
#- GPU models and configuration:
#- How you installed Lightning(`conda`, `pip`, source):
#- Running environment of LightningApp (e.g. local, cloud):
```

</details>

### More info

_No response_

### PR DESCRIPTION
**Pull request by leng-yue**
## What does this PR do?

This pull request has made changes to the Fabric and PyTorch FSDP Plugins, enabling them to handle mixed-precision training properly.

Fixes #17506

<details>
  <summary><b>Before submitting</b></summary>

- [x] Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

### CODE REVIEW

File: src/lightning/pytorch/CHANGELOG.md [hunk 1]
```diff
@@ -57,6 +57,8 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Updated `LearningRateMonitor` to log monitored values to `trainer.callback_metrics` ([#17626](https://github.com/Lightning-AI/lightning/pull/17626))
 
 
+- Fixed inconsistent settings for FSDP Precision ([#17670](https://github.com/Lightning-AI/lightning/issues/17670))
```
Comment (carmocca) [inline src/lightning/pytorch/CHANGELOG.md:60]: Can you add this also to `src/lightning/fabric/CHANGELOG.md`? Thanks

**Review by carmocca — APPROVED** [review]

**Review by Borda — APPROVED** [review]
